### PR TITLE
NMEA Fixes

### PR DESCRIPTION
**Pull request by denniszollo**
NMEA Fixes:
Add in ZDA message
Fix COG as 0 - 360
Populate Age of corrections as station ID

TODO:  Document station ID (lower bits)
             Document GEOD offset
             Add in setting for the rate of the tracking status message equivalent in NMEA
             fix RTK DOP https://github.com/swift-nav/piksi_firmware/blob/master/src/solution.c#L185

### CODE REVIEW

**Comment by denniszollo** [comment]
I'm having trouble deciding how to populate a real DOP for the RTK pseudo-absolute position here: https://github.com/swift-nav/piksi_firmware/blob/master/src/solution.c#L185 both mathematically and form a software architecture perspective.

- Possible options: call calc PVT again on the midpoint with a subset of sats and no raim
- send the SPP DOP and document that it is wrong
- bring out the DOPS as part of the baseline solving stuff (might still involve another calc_pvt)

I was hoping that our NMEA outputs would be "complete" with the next release (hence this PR).  

Anyone have some advice?
@fnoble @scottswift @benjamin0 @jacobmcnamee

**Comment by ljbade** [comment]
Shouldn't the DOP be computed from the final set of satellites used in the RTK baseline?

**Comment by denniszollo** [comment]
@ljbade that would be the correct way to compute the dop, but it would require hella firmware and libswiftnav refactoring.  Instead I'm going to use the SPP DOPS but clearly put that the DOPS isn't strictly correct in our documentation and release notes.    It's still better than a hard-coded value.

File: src/nmea.c [hunk 1]
```diff
@@ -29,8 +30,10 @@ const char NMEA_MODULE[] = "nmea";
 
 u32 gpgsv_msg_rate = 10;
 u32 gprmc_msg_rate = 10;
-u32 gpvtg_msg_rate = 10;
```
Comment (denniszollo) [inline src/nmea.c]: Course over ground at full message rate now

**Comment by denniszollo** [comment]
This is ready for code review @jacobmcnamee @gsmcmullin : 

I am a little concerned about addressing the tracking global state outside of the tracking thread. @jacob, can I do this?

I ended up using the SPP DOP and will note that in the documentation. 

It requires documentation update in our new NMEA wiki page, as well as settings.yaml for the console.

**Comment by ljbade** [comment]
Hmm interesting, we should have a dop fiction that just takes a receiver position and a set of ephemeris

**Comment by denniszollo** [comment]
TODO: get the RMC message to reflect current fix mode

**Comment by jacobmcnamee** [comment]
lgtm

**Comment by denniszollo** [comment]
@jacobmcnamee 
I added in a lock of tracking channels to avoid sending out false data - let me know if you see any issues with this.  I ran some HITL tests of this and the lock didn't seem to cause any errors.

**Comment by jacobmcnamee** [comment]
lgtm, though eventually I would like to refactor the tracking channel interface such that locks are only used internally. E.g. there could be a function which atomically populates an array with all actively tracked SIDs.

**Pull request merged**